### PR TITLE
Review comments for "Interface definition"

### PR DESCRIPTION
**Pull request by jack-oquin**
This pull request is for reviewing the referenced ROS 2 design document:
- http://design.ros2.org/articles/interface_definition.html
- https://github.com/jack-oquin/design/blob/interface_definition/articles/110_interface_definition.md

Relevant discussions:
- #48
- [REP, REP](https://groups.google.com/forum/#!topic/ros-sig-ng-ros/RaA617pJZtU)
- [Supporting both ROS versions with common sources](https://groups.google.com/forum/#!topic/ros-sig-ng-ros/61C8GUaBfKE)
- [Keeping same message package name](https://groups.google.com/forum/#!topic/ros-sig-ng-ros/wP__Hu9O8kM)

### CODE REVIEW

File: articles/110_interface_definition.md [hunk 1]
```diff
@@ -70,11 +70,15 @@ The following primitive types are defined:
...
 <div class="alert alert-warning" markdown="1">
+  **Question:** in ROS 1, `string` was ASCII. The Python 3 `str` type is UTF-8. What is the DDS `string` type?
+</div>
+
+<div class="alert alert-warning" markdown="1">
   <b>TODO:</b> consider <code>wchar</code>, <code>wstring</code>, <code>u16string</code>, <code>u32string</code>
 </div>
 
 <div class="alert alert-warning" markdown="1">
-  <b>TODO:</b> consider removing <code>byte</code>, <code>char</code> after specifying the mapping to C++ and Python
```
Comment (jack-oquin) [inline articles/110_interface_definition.md:77]: I went ahead with this change. Those types have been deprecated for many years, and the mapping was already clearly specified:
- `char`: deprecated alias for `uint8`
- `byte`: deprecated alias for `int8`

See: http://wiki.ros.org/msg#Fields

File: articles/110_interface_definition.md [hunk 2]
```diff
@@ -70,11 +70,15 @@ The following primitive types are defined:
 * `string`
 
 <div class="alert alert-warning" markdown="1">
+  **Question:** in ROS 1, `string` was ASCII. The Python 3 `str` type is UTF-8. What is the DDS `string` type?
```
Comment (dirk-thomas) [inline articles/110_interface_definition.md:73]: In ROS 1 as well as in ROS 2 the string message type is only storing a sequence of bytes. Any encoding is user land responsibility.

Comment (jack-oquin) [inline articles/110_interface_definition.md:73]: Yes, I guess that's true, but not very satisfying as a specification.

Should we recommend these bytes contain either ASCII or a UTF-8 encoding?

**Comment by jack-oquin** [comment]
This discussion seems moot at this point.

**Pull request closed without merging**